cypress/integration/login.js

```diff
@@ -17,22 +9,45 @@ describe('Sourcebooks login', function() {
     ];
 
     function makeid(length) {
-        var result = '';
-        var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
-        var charactersLength = characters.length;
-        for (var i = 0; i < length; i++) {
-           result += characters.charAt(Math.floor(Math.random() * charactersLength));
-        }
-        return result;
+        return 'Random_Name_' + Math.floor(Math.random() * 1000000000000);
     }
 
-    it('Should be able to create new task as Admin', function () {
+    it('Should display validation for empty user after attempted loggin', function () {
+        
+        cy.visit('/');
+        cy.get('.Select.not-valid').should('not.visible')
+        cy.get('[type="submit"]').click();
+        cy.get('.Select.not-valid').should('be.visible')
+    })
+
+    function testLogIn(role) {
         cy.visit('/');
         cy.get('[id="loginForm.userId"]').click({force:true});
         cy.get('[aria-label="Marius Lastauskas"]').click();
         cy.get('[id="loginForm.role"]').click({force:true});
-        cy.get('[aria-label="Admin"]').click();
+        cy.get('[aria-label="' + role + '"]').click();
+        cy.get('[type="submit"]').click();
+    }
+
+    it('Should be available for Admin to create new client', function () {
+        testLogIn('Admin');
+
+        let organization = makeid(20);
+        cy.visit('/clients');
+        cy.get('button').contains('Create Client').click();
+        cy.get('[id="clientDetailsForm.organization"]').type(organization);
+        cy.get('[id="clientDetailsForm.contacts_firstName_0"]').type(makeid(20));
+        cy.get('[id="clientDetailsForm.contacts_lastName_0"]').type(makeid(20));
+        cy.get('[id="clientDetailsForm.contacts_email_0"]').type(makeid(20) + '@mail.com');
         cy.get('[type="submit"]').click();
+        
+        cy.visit('/clients');
+        cy.get('[class="field--filter"]').first().find('input').type(organization);
+        cy.get('[class="ag-body-viewport-wrapper"').find('.ag-body-viewport').should('have.length', 1);
+    })

```

Comment:
Overall your client is consisting of bunch of numbers, it would be wise to at least prefix firstName+makeid() and so on.
Also aserting for created values would be nice too in order to confirm customer is created with expected values